internal/bootstrap/bootstrap_provider.go

```diff
@@ -30,16 +30,19 @@ import (
 	"github.com/fluxcd/go-git-providers/gitprovider"
 
 	"github.com/fluxcd/flux2/internal/bootstrap/git"
+	"github.com/fluxcd/flux2/internal/bootstrap/provider"
 	"github.com/fluxcd/flux2/pkg/manifestgen/sourcesecret"
 	"github.com/fluxcd/flux2/pkg/manifestgen/sync"
 )
 
 type GitProviderBootstrapper struct {
 	*PlainGitBootstrapper
 
-	owner      string
-	repository string
-	personal   bool
+	owner          string
+	repositoryName string
+	repository     gitprovider.UserRepository
```

Comment:
As the `repository` embeds a `provider`, I think it would be better to "lazy load" the repository (instead of making it a configuration option), to ensure one can't configure a `WithProviderUserRepository` that differs from `provider`.

Reply:
I have removed the configuration option. I take the opportunity to save at reconciliation time as we get all needed information at that moment. Later modification are not relevant. Am I getting it right?

Reply:
That assumes that the steps are forever going to be that way, which could result in `nil` panics.

Reply:
Right. Well even if we add steps, the only ones that use `repository` will be after reconciliation. And if it is set to nil, we loose the information needed to load it(when it comes to stash). I can guard against nil...